lib/rspec/mocks/object_reference.rb

```diff
@@ -41,45 +41,70 @@ def self.anonymous_module?(mod)
 
     # Used when an object is passed to `object_double`.
     # Represents a reference to that object.
-    #
-    # @private
     class DirectObjectReference
+      # @param object to which this refers to.
       def initialize(object)
         @object = object
       end
 
+      # The object's description (via `#inspect`).
+      # @return String
       def description
         @object.inspect
       end
 
+      # Defined for interface parity with the other object reference
+      # implementations, it has no real relevance here.
       def const_to_replace
         raise ArgumentError,
               "Can not perform constant replacement with an object."
       end
 
+      # The target of the verifying double (the object's class).
+      #
+      # @return Object
       def target
-        @object.class
+        @object
       end
 
+      # Always returns true for an object as the class is defined.
+      #
+      # @return true
       def defined?
         true
       end
 
+      # Yields if the reference target is loaded, providing a generic mechanism
+      # to optionally run a bit of code only when a reference's target is
+      # loaded.
+      #
+      # This specific implementation always yields because direct references
+      # are always loaded.
+      #
+      # @yield object
       def when_loaded
         yield @object
       end
     end
 
-    # Used when an anonymous module is passed to `class_double` or `instance_double`.
-    # Represents a reference to that module.
-    #
-    # @private
+    # Used when an anonymous module is passed to `class_double` or
+    # `instance_double`. Represents a reference to that module.
     class AnonymousModuleReference < DirectObjectReference
+      # Returns the constant name to replace with a double.
+      #
+      # @return String the constant name
       def const_to_replace
         @object.name
       end
+
+      # Returns the module name.
+      #
+      # @return String the constant name
       alias description const_to_replace
```

Comment:
I think YARD automatically docs aliases w/o needing doc comments (in fact, we hook into that for our matcher) aliases.  See http://rspec.github.io/documentation/3.2/rspec-expectations/RSpec/Matchers.html for an example. Each matcher lists its aliases using `(also: #some_alias)` and I believe it would do the same here but you'd have to check.